File: solutions/go/hamming/1/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
    if len(a) != len(b) {
        return 0, errors.New("not equal")
    }
    length := len(a)
    var count int = 0;
    for i := 0; i < length; i++ {
        if (a[i] != b[i]) {
            count += 1
        }
    }
    return count, nil
}

File: solutions/c/hamming/1/hamming.c
#include "hamming.h"

#include <string.h>

int compute(const char *lhs, const char *rhs) {
    if (strlen(lhs) != strlen(rhs)) {
        return -1;
    }
    int distance = 0;
    int i = 0;
    while (lhs[i] != '\0') {
        if (lhs[i] != rhs[i]) {
            distance++;
        }
        i++;
    }
    return distance;
}
